sros2/sros2/keystore/_keystore.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2020 Canonical Ltd
+# Copyright 2016-2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import pathlib
+
+from cryptography import x509
+
+import lxml
+
+from sros2 import _utilities
+import sros2.errors
+from sros2.policy import get_transport_default, get_transport_schema
+
+
+_KS_ENCLAVES = 'enclaves'
+_KS_PUBLIC = 'public'
+_KS_PRIVATE = 'private'
+_DEFAULT_COMMON_NAME = 'sros2CA'
+
+
+def create_keystore(keystore_path: pathlib.Path) -> None:
+    if is_valid_keystore(keystore_path):
+        raise sros2.errors.KeystoreExistsError(keystore_path)
+
+    for path in (
+            keystore_path,
+            keystore_path.joinpath(_KS_PUBLIC),
+            keystore_path.joinpath(_KS_PRIVATE),
+            keystore_path.joinpath(_KS_ENCLAVES)):
+        path.mkdir(parents=True, exist_ok=True)
+
+    keystore_ca_cert_path = keystore_path.joinpath(_KS_PUBLIC, 'ca.cert.pem')
+    keystore_ca_key_path = keystore_path.joinpath(_KS_PRIVATE, 'ca.key.pem')
+
+    keystore_permissions_ca_cert_path = keystore_path.joinpath(
+        _KS_PUBLIC, 'permissions_ca.cert.pem')
+    keystore_permissions_ca_key_path = keystore_path.joinpath(
+        _KS_PRIVATE, 'permissions_ca.key.pem')
+
+    keystore_identity_ca_cert_path = keystore_path.joinpath(
+        _KS_PUBLIC, 'identity_ca.cert.pem')
+    keystore_identity_ca_key_path = keystore_path.joinpath(
+        _KS_PRIVATE, 'identity_ca.key.pem')
+
+    required_files = (
+        keystore_permissions_ca_cert_path,
+        keystore_permissions_ca_key_path,
+        keystore_identity_ca_cert_path,
+        keystore_identity_ca_key_path,
+    )
+
+    # Create new CA if one doesn't already exist
+    if not all(os.path.isfile(x) for x in required_files):
```

Comment:
Same here, if they are all pathlib.Path we can use function from pathlib instad of os.path
```suggestion
    if not all(x.is_file()) for x in required_files):
```

Reply:
Thank you, fixed.